src/commands/npm/release/validate.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { flags, FlagsConfig, SfdxCommand } from '@salesforce/command';
+import { isMonoRepo, LernaRepo } from '../../../repository';
+import { Package } from '../../../package';
+import { CommitInspection, inspectCommits } from '../../../inspectCommits';
+
+type PackageCommits = CommitInspection & {
+  name: string;
+  currentVersion: string;
+};
+
+type Response = {
+  shouldRelease: boolean;
+  packages?: PackageCommits[];
+};
+
+export default class Validate extends SfdxCommand {
+  public static readonly description =
+    'inspects the git commits to see if there are any commits that will warrant a new release';
+  public static readonly flagsConfig: FlagsConfig = {
+    verbose: flags.builtin({
+      description: 'show all commits for all packages (only works with --json flag)',
+    }),
+  };
+
+  public async run(): Promise<Response> {
+    const isLerna = await isMonoRepo();
+    const packages = isLerna ? await LernaRepo.getPackages() : [await Package.create()];
+    const responses: PackageCommits[] = [];
+    for (const pkg of packages) {
+      const commitInspection = await inspectCommits(pkg, isLerna);
+      const response = Object.assign(commitInspection, {
+        name: pkg.name,
+        currentVersion: pkg.packageJson.version,
+      });
+      responses.push(response);
+    }
+    const shouldRelease = responses.some((resp) => !!resp.shouldRelease);
+    this.ux.log(shouldRelease.toString());
+    return this.flags.verbose ? { shouldRelease, packages: responses } : { shouldRelease };
```

Comment:
@mdonnalley thanks for jumping on this! This will require examination of the console output to calculate if the job should continue. Would it be possible to also change the exit code so interpretation of the result is simpler?

Reply:
I agree, if we shouldn't release the exit code should be 1 to stop CI

Reply:
I'm fine if this exits with 1, but I don't think we want the CI job to fail. Otherwise it will report failed builds when it isn't really a failure. It just shouldn't do it.

Reply:
@amphro , the change that needs to be applied to the orb will have to capture the return code and then act accordingly.